### PR TITLE
Improve test setup

### PR DESCRIPTION
**Pull request by koesie10**
Please review this commit-by-commit.

This creates a new test suite, updates the script names and updates the documentation. The new test suite is essentially the same as the CLI integration test suite, but does not require the CodeQL repository to be added to the workspace and is not run against multiple versions in CI.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

**Review by elenatanasoiu — APPROVED** [review]
Looks good!

**Review by charisk — COMMENTED** [review]
Sorry I started reviewing last week but got distracted!

Some comments - mostly questions.

File: extensions/ql-vscode/test/vscode-tests/jest.activated-extension.setup.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+beforeAll(async () => {
+  // Set the CLI version here before activation to ensure we don't accidentally try to download a cli
+  await beforeEachAction();
+  await CUSTOM_CODEQL_PATH_SETTING.updateValue(
+    process.env.CLI_PATH,
+    ConfigurationTarget.Workspace,
+  );
+
+  // ensure the test database is downloaded
```
[the remark this answers is not in the document]
Comment (charisk) [reply extensions/ql-vscode/test/vscode-tests/jest.activated-extension.setup.ts:26]: BTW if it'll make the implementation complicated to remove it just for the activated tests without much benefit (not sure how much time it takes), then I wouldn't mind us leaving it here. Perhaps we can just add a comment to explain.

File: extensions/ql-vscode/test/vscode-tests/activated-extension/jest-runner-vscode.config.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "github.codespaces",
+    "--disable-extension",
+    "github.copilot",
+    path.resolve(rootDir, "test/data"),
+  ],
+  extensionTestsEnv: {
+    ...baseConfig.extensionTestsEnv,
+    INTEGRATION_TEST_MODE: "true",
+  },
+  retries: 3,
```
[the remark this answers is not in the document]
Comment (koesie10) [reply extensions/ql-vscode/test/vscode-tests/activated-extension/jest-runner-vscode.config.js:26]: I think the flakiness does come from the test infrastructure and needing to start VSCode. Sometimes, VSCode doesn't seem to run all the tests, so we need retries to finish all tests. See https://github.com/github/vscode-codeql/pull/1801 and https://github.com/github/vscode-codeql/blob/682a1e12bc9d4d157cbfe35b757e71a5f076cee9/extensions/ql-vscode/patches/jest-runner-vscode%2B3.0.1.patch#L111-L116.

File: extensions/ql-vscode/test/vscode-tests/jest.activated-extension.setup.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (koesie10) [reply extensions/ql-vscode/test/vscode-tests/jest.activated-extension.setup.ts:26]: No, we don't really need this as part of the activation, so I'll move this so it only occurs for CLI integration tests.

File: extensions/ql-vscode/test/vscode-tests/activated-extension/jest-runner-vscode.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (charisk) [reply extensions/ql-vscode/test/vscode-tests/activated-extension/jest-runner-vscode.config.js:26]: Ok thanks!

**Review by charisk — APPROVED** [review]

**Pull request merged**